### PR TITLE
New reference doc for Custom RPC V2

### PR DESCRIPTION
**Pull request by kianenigma**
Thanks for @xlc for the original seed info, I've just fixed it up a bit and added example links. 

I've moved the comparison between eth-rpc-api and frontier outside, as it is opinionation. I think the content there was good but should live in the README of the corresponding repos. No strong opinion, happy either way.

### CODE REVIEW

File: docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+//!
+//! Finally, this custom RPC needs to be integrated into the node side. This is usually done in a
+//! `rpc.rs` in a typical template, as follows:
+//!
+#![doc = docify::embed!("../../templates/minimal/node/src/rpc.rs", create_full)]
+//!
+//! ## Future
+//!
+//! [XCQ](https://forum.polkadot.network/t/cross-consensus-query-language-xcq/7583) will be a good
+//! solution for most of the query needs.
```
Comment (kianenigma) [inline docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs:79]:
```suggestion
//! - [XCQ](https://forum.polkadot.network/t/cross-consensus-query-language-xcq/7583) will be a good
//! solution for most of the query needs.
//! - [New JSON-RPC Specification](https://github.com/paritytech/json-rpc-interface-spec)
```

**Review by bkchr — APPROVED** [review]

File: docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+//! # Custom RPC do's and don'ts
+//!
+//! **TLDR:** don't create new custom RPCs. Instead, rely on custom Runtime APIs, combined with
+//! `state_call`
+//!
+//! ## Background
+//!
+//! Polkadot-SDK offers the ability to query and subscribe storages directly. However what it did
+//! not have is [view functions](https://github.com/paritytech/polkadot-sdk/issues/216). This is an
```
Comment (gpestana) [inline docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs:9]:
```suggestion
//! Polkadot-SDK offers the ability to query and subscribe storages directly. However what it does
//! not have is [view functions](https://github.com/paritytech/polkadot-sdk/issues/216). This is an
```

**Review by gpestana — APPROVED** [review]

File: docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs [hunk 3]
```diff
@@ -0,0 +1,80 @@
+//! # Custom RPC do's and don'ts
+//!
+//! **TLDR:** don't create new custom RPCs. Instead, rely on custom Runtime APIs, combined with
+//! `state_call`
+//!
+//! ## Background
+//!
+//! Polkadot-SDK offers the ability to query and subscribe storages directly. However what it did
```
Comment (michalkucharczyk) [inline docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs:8]: would be nice to have some links here.

File: docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+//! For example, let's take a look a the process through which the account nonce can be queried
+//! through an RPC. First, a new runtime-api needs to be declared:
+//!
+#![doc = docify::embed!("../../substrate/frame/system/rpc/runtime-api/src/lib.rs", AccountNonceApi)]
+//!
+//! This API is implemented at the runtime level, always inside [`sp_api::impl_runtime_apis!`].
+//!
+//! As noted, this is already enough to make this API usable via `state_call`.
+//!
+//! ## Create a new custom RPC
```
Comment (michalkucharczyk) [inline docs/sdk/src/reference_docs/custom_runtime_api_rpc.rs:62]: Maybe those two sections:
## Create a new custom RPC
## Add a new RPC to the node

should be placed under one root section as they are dependent on each other (IIUC).

**Review by michalkucharczyk — APPROVED** [review]

**Review by kianenigma — COMMENTED** [review]
mark as legacy

**Comment by kianenigma** [comment]
bot fmt

**Review by lexnv — APPROVED** [review]

**Pull request merged**